src/uri/UriValidator.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <uprotocol-cpp/uri/validator/UriValidator.h>
+#include "uprotocol/uri.pb.h"
+#include "uprotocol/ustatus.pb.h"
+#include <uprotocol-cpp/uri/serializer/LongUriSerializer.h>
+#include <uprotocol-cpp/uri/serializer/MicroUriSerializer.h>
+#include <string.h>
+
+using namespace uprotocol::uri;
+using namespace uprotocol::v1;
+
+/**
+ * isValidUri a {@link uri} to ensure that it has at least a name for the
+ * uEntity.
+ *
+ * @param uri {@link uri} to validate.
+ * @return Returns bool containing a true or a false
+ */
+bool UriValidator::isValidUri(const std::string& uri) {
+    auto uri_view = LongUriSerializer::deserialize(uri);
+    return !isEmpty(uri_view);   
+}
+
+/**
+ * Validate a {@link UUri} to ensure that it has at least a name for the uEntity.
+ *
+ * @param uri {@link UUri} to validate.
+ * 
+ * @return Returns Status containing a success or a failure with the error message.
+ */
+UStatus UriValidator::validate(const UUri& uri) {
+  UStatus status;
+  if ((nullptr == &uri) || isEmpty(uri)) {
+  status.set_code(UCode::INVALID_ARGUMENT);
+    status.set_message("Uri is empty.");
+    return status;
+  }
+  if (uri.has_authority() && !isRemote(uri.authority())) {
+    status.set_code(UCode::INVALID_ARGUMENT);
+    status.set_message("Uri is remote missing uAuthority.");
+    return status;
+  }
+  if (!isEmpty(uri.entity())  && isBlank( uri.entity().name())) {
+    status.set_code(UCode::INVALID_ARGUMENT);
+    status.set_message("Uri is missing uSoftware Entity name.");
+    return status;
+  }
+  status.set_code(UCode::OK);
```

Comment:
This then would be the else case.  This allows the function to have only one return which is favorable for readability.

Reply:
Done